test_project/simple_api_test/testcases/objects.py

```diff
@@ -1,68 +1,31 @@
-from django.contrib.auth.models import User as UserModel
-
 from simple_api.adapters.graphql.graphql import GraphQLAdapter
 from simple_api.adapters.utils import generate
-from simple_api.django_object.actions import CreateAction, UpdateAction, DeleteAction, DetailAction, ListAction, ModelAction
 from simple_api.django_object.django_object import DjangoObject
-from simple_api.django_object.permissions import IsAuthenticated
-from simple_api.object.datatypes import StringType
-from simple_api.object.permissions import Or
+from simple_api.object.actions import Action
+from simple_api.object.datatypes import IntegerType, ObjectType
+from simple_api.object.validators import FieldValidator
 
 from tests.graphql.graphql_test_utils import build_patterns
 
-from .models import Post as PostModel
-
-
-class IsAdmin(IsAuthenticated):
-    def permission_statement(self, request, obj, **kwargs):
-        return request.user.is_staff or request.user.is_superuser
-
-
-class IsSelf(IsAuthenticated):
-    def permission_statement(self, request, obj, **kwargs):
-        return request.user == obj
-
-
-class IsTheirs(IsAuthenticated):
-    def permission_statement(self, request, obj, **kwargs):
-        return request.user == obj.author
-
-
-def set_password(request, params, **kwargs):
-    # this would set a password to request.user, but for the showcase it is not needed
-    pass
-
-
-def create_post(request, params, **kwargs):
-    data = params["data"]
-    return PostModel.objects.create(title=data["title"], content=data["content"], author=request.user)
+from .models import Book as BookModel
 
 
-class User(DjangoObject):
-    model = UserModel
-    only_fields = ("username", )
+def IsPositive(value, **kwargs):
+    print("WTF?", value)
+    print(kwargs)
+    return value >= 0
 
-    create_action = CreateAction(permissions=IsAdmin)
-    update_action = UpdateAction(permissions=IsAdmin)
-    delete_action = DeleteAction(permissions=IsAdmin)
-    detail_action = DetailAction(permissions=IsAdmin)
-    list_action = ListAction(permissions=IsAdmin)
 
-    custom_actions = {
-        "changePassword": ModelAction(data={"password": StringType()}, exec_fn=set_password,
-                                      permissions=IsAuthenticated),
-        "myProfile": ModelAction(exec_fn=lambda request, **kwargs: request.user, permissions=IsAuthenticated)
-    }
+def get_by_id(request, params, **kwargs):
+    return BookModel.objects.get(id=params["id"])
 
 
-class Post(DjangoObject):
-    model = PostModel
+class Book(DjangoObject):
+    IsPositiveValidator = FieldValidator(IsPositive)
```

Comment:
toto vubec nechapu, myslel jsem, ze jsme si rozumeli, zkusim byt explicitnejsi: napises si classu pro validator

```python
class IsPositiveValidator(Validator):
   def validate_statement(self, value, **kwargs):
       return value > 0
```

a potom udelas

```python
parameters={"id": IntegerType(validators=IsPositiveValidator)}
```

tj. do objektu kvuli tomu nepridavas zadne dalsi class atributy

Reply:
Jasně, přetáhnu tu funkci do validate_statement, jen jsem to už měl jako argument tak jsem to na testování v tom nechal, bude vyřešeno v dalším commitu